ducktape/cluster/json.py

```diff
@@ -52,33 +79,52 @@ def __init__(self, cluster_json=None, *args, **kwargs):
 
         self.available_nodes = collections.deque(node_accounts)
         self.in_use_nodes = set()
-        self.id_source = 1
 
     def __len__(self):
         return len(self.available_nodes) + len(self.in_use_nodes)
 
     def num_available_nodes(self):
         return len(self.available_nodes)
 
-    def request(self, nslots):
-        if nslots > self.num_available_nodes():
+    def alloc_subcluster(self, num_nodes):
+        nodes = self.alloc(num_nodes)
+        return JsonCluster(
```

Comment:
Do we need each subcluster to actually use the same cluster type? I figured we'd just `alloc` from the cluster, then use the result to create a generic cluster since they all just need the same key properties (as in `_account_to_dict`. Then the subcluster part isn't anything special from the perspective of the `Cluster` implementation, you still just `alloc` and `free` sets of nodes.

I think only the freeing step is what you need the original set of nodes for. In that case, couldn't we just make whatever class implements `Subcluster` accept the original set of nodes in its constructor so `free_subcluster` below can use the original set rather than "allocating" them back from the subcluster?